daffodil-japi/src/main/java/org/apache/daffodil/japi/package-info.java

```diff
@@ -151,11 +151,11 @@
  * InputSourceDataInputStream is = new InputSourceDataInputStream(dataStream);
  * JDOMInfosetOutputter jdomOutputter = new JDOMInfosetOutputter();
  * boolean keepParsing = true;
- * while (keepParsing) {
+ * while (keepParsing && is.isDefinedForLength(1)) {
```

Comment:
Thoughts on onsidering a differenent public API function that gives a little less power to the user? This functions let's the user prefetch anymount they want, e.g. `is.isDefinedForLength(500000)``. Which means we'll block until 500000 bits are cached. Any concerns with letting a user do that?

An alternative is maybe something like ``is.hasData()`` which could essentially be a public alias for ``is.isDefinedForLength(1)``, but limits what a user could do. We could always make isDefinedForLength public in the future if users did need it, but it's harder to take it away wihtout going through a deprecation process.

Reply:
Good suggestion. I will add this.